qiskit/transpiler/basepasses.py

```diff
@@ -95,6 +95,45 @@ def is_analysis_pass(self):
         """
         return isinstance(self, AnalysisPass)
 
+    def __call__(self, circuit, property_set=None):
+        """Runs the pass on circuit.
+
+        Args:
+            circuit (QuantumCircuit): the dag on which the pass is run.
+            property_set (PropertySet or dict or None): input/output property set.
+
+        Returns:
+            QuantumCircuit: If on transformation pass, the resulting QuantumCircuit. If analysis
+                   pass, the input circuit.
+        """
+        from qiskit.converters import circuit_to_dag, dag_to_circuit
+        from qiskit.dagcircuit.dagcircuit import DAGCircuit
+
+        property_set_ = None
+        if isinstance(property_set, dict):
+            property_set_ = PropertySet(property_set)
+
+        if isinstance(property_set_, PropertySet):
+            self.property_set = property_set_
```

Comment:
```suggestion
        property_set_ = None
        if isinstance(property_set, dict):
            property_set_ = PropertySet(property_set)

        if isinstance(property_set, PropertySet):
            self.property_set = property_set
```
I'm not 100% sure I follow the cases here, but it looks like if `property_set` is given as a `PropertySet`, it won't be set on `self` before `self.run`.

Reply:
oh! see the confusion. It is because `isinstance(PropertySet(), dict) == True`.

I clarified that in 98d53856bd77b23282c16169829d77d096a17847 . Have a look.